File: backends/java/zappy-host/src/main/java/com/zappymart/host/ZappyMartApplication.java
package com.zappymart.host;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.persistence.autoconfigure.EntityScan;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;

@SpringBootApplication(scanBasePackages = "com.zappymart")
@EntityScan("com.zappymart.adapters.persistence")
@EnableJpaRepositories("com.zappymart.adapters.persistence")
public class ZappyMartApplication {

    public static void main(String[] arguments) {
        SpringApplication.run(ZappyMartApplication.class, arguments);
    }
}

File: backends/kotlin/zappy-host/src/main/kotlin/nl/zappymart/host/ZappyMartApplication.kt
package nl.zappymart.host

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.persistence.autoconfigure.EntityScan
import org.springframework.boot.context.properties.ConfigurationPropertiesScan
import org.springframework.boot.runApplication
import org.springframework.context.annotation.ComponentScan
import org.springframework.data.jpa.repository.config.EnableJpaRepositories

@SpringBootApplication
@ComponentScan(basePackages = ["nl.zappymart.host", "nl.zappymart.adapters"])
@ConfigurationPropertiesScan(basePackages = ["nl.zappymart.adapters"])
@EntityScan(basePackages = ["nl.zappymart.adapters.persistence.entities"])
@EnableJpaRepositories(basePackages = ["nl.zappymart.adapters.persistence"])
class ZappyMartApplication

fun main(arguments: Array<String>) {
    runApplication<ZappyMartApplication>(*arguments)
}
